cores/portduino/main.cpp

```diff
@@ -75,12 +78,18 @@ static error_t parse_opt(int key, char *arg, struct argp_state *state) {
 }
 
 void getMacAddr(uint8_t *dmac) {
-  dmac[0] = 0x80;
-  dmac[1] = 0;
-  dmac[2] = portduinoArguments.hwId >> 24;
-  dmac[3] = portduinoArguments.hwId >> 16;
-  dmac[4] = portduinoArguments.hwId >> 8;
-  dmac[5] = portduinoArguments.hwId & 0xff;
+    std::fstream macIdentity;
+    macIdentity.open("/sys/kernel/debug/bluetooth/hci0/identity", std::ios::in);
```

Comment:
There should be a way do retrieve the bluetooth address without root rights (e.g. see hciconfig). meshtaticd should not require sudo permissions for running.

Reply:
This now runs without root, and fails gracefully.